### PR TITLE
bug 1490727: Feature/implement recurring payment designs

### PR DESCRIPTION
**Pull request by joshJarr**
**DO NOT MERGE THIS PULL REQUEST - THIS IS FOR REVIEW PURPOSES ONLY**
This is a cleaned version of https://github.com/potatolondon/kuma/pull/35 with only the frontend changes.

-----

Done some refactoring in the styles and included the flow for the recurring payments. The popover banner should not display the new UI and work as usual. The new UI can be accessed on the `/payments` view, with the recurring payments at `/payments/recurring`.
Design is implemented and passed QA.

**In this PR**

- bug 1490727: Updated media queries to tablet and mobile displays fit. (https://trello.com/c/NDTSpBwz/37-update-tablet-breakpoints-for-the-contribution-banner)
         - Updated media queries to tablet and mobile displays fit.
         - Solve margin layout on label inputs

- https://trello.com/c/FArWmeK0/2-as-a-user-i-can-select-either-one-time-or-recurring-so-that-i-can-decide-the-way-i-want-to-pay
 
- https://trello.com/c/fFLk8AEi/4-as-a-user-i-can-choose-the-amount-that-i-want-to-set-for-recurring-payments-so-that-i-can-pay-the-amount-i-can-afford

- https://trello.com/c/iDMBjyk6/9-as-a-user-i-need-to-be-able-to-enter-my-details-for-recurring-payments-so-that-i-can-set-up-my-recurring-payments

- (sprint 2) https://trello.com/c/krA4aQqY/6-as-a-user-i-can-see-the-payment-options-for-recurring-payments-so-that-i-can-see-if-my-payment-option-is-accepted

**Notes for QA**
- [ ] ensure popover works as usual
- [ ] ensure the popover animation works as expected
- [ ] ensure banner displays on the FAQ page correctly when feature is enabled.
- [ ] ensure the FAQ page works as usual when the recurring payment flag is disabled
- [ ] ensure the user cannot hit `/payments/recurring` when the recurring payments flag is disabled
- [ ] ensure the new UI and recurring payments displays when the waffle flag is enabled
- [ ] ensure the recurring payment form looks correctly across browsers and devices
- [ ] ensure recurring payment form validation works
- [ ] ensure recurring payment form can be submitted
- [ ] ensure details are saved in stripe correctly
- [ ] ensure recurring payment tabs redirect the user correctly
- [ ] ensure that being logged into MDN displays your user name and email in the form field by default
- [ ] ensure you can edit your user detail in the payment form if they are pre-filled
- [ ] ensure using stripe's test error cards correctly triggers validation or redirects to an error page upon submission https://stripe.com/docs/testing

**Screenshot of changes**

screen shot 2018-10-31 at 12 03 09
screen shot 2018-10-31 at 12 02 52
screen shot 2018-10-31 at 12 02 40
screen shot 2018-10-31 at 12 02 22

### CODE REVIEW

File: kuma/static/styles/components/payments/banner.scss [hunk 1]
```diff
@@ -118,12 +138,75 @@
...
+        max-width: $payment-banner-max-width;
+        padding: $payment-banner-padding;
         position: relative;
+        transition: padding .3s ease;
+
+        > .column-half {
+            box-sizing: border-box;
+
+            .container {
+                max-width: $payment-column-max-width;
```
Comment (matthewhall) [inline kuma/static/styles/components/payments/banner.scss]: Alphabetise these please. And elsewhere as well.

File: kuma/static/styles/components/payments/form.scss [hunk 2]
```diff
@@ -116,17 +72,64 @@
...
+            @media #{$mq-large-desktop-and-down} {
+                li {
+                    margin: 0;
+                }
+            }
+        }
+
+        @media #{$mq-mobile-and-down} {
+            .form-radios-donation-choices {
+                margin-right: -3px; // ensure % fits
```
Comment (matthewhall) [inline kuma/static/styles/components/payments/form.scss]: Not an issue for now, but could we avoid doing this if we used flexbox?

File: kuma/payments/jinja2/payments/includes/payments-form.html [hunk 3]
```diff
@@ -1,114 +1,67 @@
...
-                    <div class="button-group">
-                        <a href="{{ faq_url }}" class="cta-link">
-                            {{_('Learn more about payments to MDN')}} {% include 'includes/icons/arrows/arrow.svg' %}
-                        </a>
-                        <a href="{{ feedback_url }}" class="cta-link">
-                            {{_('Provide feedback')}} {% include 'includes/icons/arrows/arrow.svg' %}
-                        </a>
-                    </div>
-                {% endif %}
+<form id="contribute-form" class="contribution-form" method="{% if recurring_payment and not request.user.is_authenticated %}get{% else %}post{% endif %}" action="{% if recurring_payment %}{{url('recurring_payment_subscription')}}{%else%}{{url('payments')}}{%endif%}" novalidate>
```
Comment (matthewhall) [inline kuma/payments/jinja2/payments/includes/payments-form.html]: Some inconsistent whitespace on this line: `{%else%}`.

File: kuma/static/styles/components/payments/_vars.scss [hunk 4]
```diff
@@ -1,6 +1,26 @@
...
-$contribution-background-gradient: linear-gradient(to right, nth($contribution-background-colors, 1) , nth($contribution-background-colors, 2));
+// Width of banner containers
+$payment-banner-max-width: 1320px;
+$payment-column-max-width: 540px;
 
-$contribution-form-input-height: 56px;
-$contribution-form-input-spacing: 18px;
-$contribution-form-button-color: #3143e2;
+// Height of the popover when expanded
+$payment-banner-popover-height: 480px;
```
[the remark this answers is not in the document]
Comment (joshJarr) [reply kuma/static/styles/components/payments/_vars.scss]: Yeah I agree, We have a bug with the banner opening animation, will tackle this issue with that 👍

File: kuma/static/styles/components/payments/form.scss (hunk 2 above)
Comment (joshJarr) [reply kuma/static/styles/components/payments/form.scss]: 👍 I had planned on doing this when we have time to refactor.

File: kuma/landing/jinja2/landing/homepage.html [hunk 5]
```diff
@@ -153,6 +153,7 @@ <h2>{% include 'includes/icons/general/smile.svg' %} {{ _('Help improve MDN') }}
     {% javascript 'newsletter' %}
   {% endif %}
   {%- if contribution_popup %}
+    {% javascript 'tooltip' %}
```
[the remark this answers is not in the document]
Comment (joshJarr) [reply kuma/landing/jinja2/landing/homepage.html]: Addressed.

**Pull request closed without merging**